aiokafka/protocol/struct.py

```diff
@@ -23,27 +25,29 @@ def __init__(self, *args, **kwargs):
                     )
                 )
 
-    def encode(self):
+    def encode(self) -> bytes:
         return self.SCHEMA.encode([self.__dict__[name] for name in self.SCHEMA.names])
 
     @classmethod
-    def decode(cls, data):
+    def decode(cls, data: Union[BytesIO, bytes]) -> Self:
         if isinstance(data, bytes):
             data = BytesIO(data)
         return cls(*[field.decode(data) for field in cls.SCHEMA.fields])
 
-    def get_item(self, name):
+    def get_item(self, name: str) -> Any:
         if name not in self.SCHEMA.names:
             raise KeyError("%s is not in the schema" % name)
         return self.__dict__[name]
 
-    def __repr__(self):
-        key_vals = []
+    def __repr__(self) -> str:
+        key_vals: List[str] = []
         for name, field in zip(self.SCHEMA.names, self.SCHEMA.fields):
             key_vals.append(f"{name}={field.repr(self.__dict__[name])}")
         return self.__class__.__name__ + "(" + ", ".join(key_vals) + ")"
 
-    def __eq__(self, other):
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, Struct):
+            return False
```

Comment:
Maybe return NotImplemented?

Reply:
Where can it make difference? When comparing with `mock.ANY`? Ok, let's return `NotImplemented`